weneige/orders/views.py

```diff
@@ -1,3 +1,53 @@
-from django.shortcuts import render
+import json
 
-# Create your views here.
+from django.core.exceptions import ValidationError
+from django.http            import JsonResponse
+from django.views           import View
+
+from .utils.utils           import login_decorator
+from .models                import Order, OrderStatus
+from products.models        import Product, ProductImage, ProductOption, Color, Volume
+class OrderView(View):
+    @login_decorator
+    def post(self, request):
+        try :
+            data        = json.loads(request.body)
+            user        = request.user
+            product_id  = data['product_id']
+            quantity    = data['quantity']
+            color_id    = data['color_id']
+
+            Order.objects.create(
+                user            = user,
+                product_option  = ProductOption.objects.filter(product_id=product_id, color_id=color_id)[0],
+                quantity        = quantity,
+                address         = user.address,
+                mobile_number   = user.mobile_number,
+                order_status    = OrderStatus.object.get(id=1).name
+            )
+        
+            return JsonResponse({'MESSAGE' : 'SUCCESS'}, status = 201)
+        
+        except KeyError:
+            return JsonResponse({'MESSAGE' : 'KEY_ERROR'}, status = 400)
+
+    @login_decorator
+    def get(self, request):
+        user        = request.user
+        user_orders = user.user_order.filter(user_id=user.id)
+        results     = []
+
+        for user_order in user_orders:
+            results.append(
+                {
+                    "kor_name"  : Product.objects.get(id=user_order.option_order.product_id).kor_name,
```

Comment:
```suggestion
                    "kor_name"  : user_order.option_order.product.kor_name,
```
43~47 line code는 다음과 같이 바꾸어도 충분히 원하는 특정 쿼리셋을 불러올 수 있습니다.

Reply:
혹시 위와 같이만 써도 되는 원리가 어떻게 될까요? models.py 에서 Field 생성시 related_name에 따로 product에 대해서 설정을 주지 않은 것 같은데 어떻게 위와 같이 사용 가능한지 알고 싶습니다.